global_helpers/global_helpers_test.py

```diff
@@ -351,6 +351,34 @@ def test_context(self):
         self.assertEqual(expected, self.ip_info.context(self.match_field))
 
 
+class TestOnlyCrowdStrikeFdrEventType(unittest.TestCase):
+    def setUp(self):
+        self.input = {
+            "p_log_type": "Crowdstrike.FDREvent",
+            "aid": "else",
+            "event": {"foo": "bar"},
+            "fdr_event_type": "DnsRequest",
+        }
+
+    def test_is_different_with_fdr_event_type_provided(self):
+        response = p_b_h.contains_crowdstrike_fdr_event_type(self.input, "SomethingElse")
+        self.assertEqual(response, False)
+
+    def test_is_same_with_the_fdr_event_type_provided(self):
+        response = p_b_h.contains_crowdstrike_fdr_event_type(self.input, "DnsRequest")
+        self.assertEqual(response, True)
+
+    def test_is_entirely_different_type(self):
+        self.input = {
+            "p_log_type": "Crowdstrike.DnsRequest",
+            "aid": "else",
+            "event": {"foo": "bar"},
+            "fdr_event_type": "DnsRequest",
```

Comment:
```suggestion
            "event_simpleName": "DnsRequest",
```

I do not see the `fdr_event_type` key/column on `Crowdstrike.DnsRequest`.

Reply:
`fdr_event_type` is a field that gets added only for the new `Crowdstrike.FDREvent` logtype.
`Crowdstrike.DnsRequest` is a legacy one.

Reply:
concur. I am struggling to understand why we would establish a test case for `p_log_type=Crowdstrike.DnsRequest` that also has a `fdr_event_type=DnsRequest`.  It looks to me that when p_log_type=Crowdstrike.DnsRequest, fdr_event_type is called `event_simpleName`

Reply:
If I'm getting Ed's comment right, this seems as an impossible input, since a processed log cannot be of `p_log_type=Crowdstrike.DnsRequest` **and** have `fdr_event_type` field populated. Is that right?

Reply:
Hmm, that is correct. Removed: https://github.com/panther-labs/panther-analysis/pull/673/commits/3b78e66ef53215ed2d06ee3b7aa20d56e40b4269
Also renamed the function and swapped the boolean logic so it would be a bit more readable.
Let's see.